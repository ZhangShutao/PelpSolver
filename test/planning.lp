step(1). step(2).
next(1, 2). next(2, 3).
time(1). time(2). time(3).
status(holding). status(dry).

-holds(X, 1) :- status(X), not holds(X, 1).
holds(X, T2) :- holds(X, T1), next(T1, T2), not -holds(X, T2), step(T1).
-holds(X, T2) :- -holds(X, T1), next(T1, T2), not holds(X, T2), step(T1).
action(pick, T) | action(bake, T) :- step(T).
:- action(pick, X), action(bake, X).

holds(holding, Y) :- holds(dry, X), -holds(holding, X), action(pick, X), next(X, Y), step(X). [3]
holds(holding, Y) :- -holds(dry, X), -holds(holding, X), action(pick, X), next(X, Y), step(X). [-1]
holds(dry, Y) :- action(bake, X), next(X, Y), step(X). [3]

-holds(holding, 1).
holds(dry, 1). [-1]

success :- holds(holding, 3).
goal :- K(0.8,1] success.
:- not goal.
:- K[0,1) action(A, S), action(A, S).