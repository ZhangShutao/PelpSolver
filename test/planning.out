WorldView 1:
{[0.030711819236122834]:{action(bake,1),-holds(holding,3),status(dry),-holds(holding,1),-holds(holding,2),step(2),status(holding),action(pick,2),goal,time(1),-holds(dry,1),time(2),time(3),step(1),holds(dry,3),next(1,2),next(2,3),holds(dry,2)},[0.030711819236122834]:{action(bake,1),status(dry),-holds(holding,1),-holds(holding,2),step(2),status(holding),action(pick,2),success,goal,-holds(dry,2),time(1),-holds(dry,1),time(2),time(3),-holds(dry,3),holds(holding,3),step(1),next(1,2),next(2,3)},[0.011298246897943221]:{action(bake,1),-holds(holding,3),status(dry),-holds(holding,1),-holds(holding,2),step(2),status(holding),action(pick,2),goal,time(1),time(2),time(3),step(1),holds(dry,1),holds(dry,3),next(1,2),next(2,3),holds(dry,2)},[0.6168633792454105]:{action(bake,1),status(dry),-holds(holding,1),-holds(holding,2),step(2),status(holding),action(pick,2),success,goal,time(1),-holds(dry,1),time(2),time(3),holds(holding,3),step(1),holds(dry,3),next(1,2),next(2,3),holds(dry,2)},[0.08348338014847165]:{action(bake,1),-holds(holding,3),status(dry),-holds(holding,1),-holds(holding,2),step(2),status(holding),action(pick,2),goal,-holds(dry,2),time(1),-holds(dry,1),time(2),time(3),-holds(dry,3),step(1),next(1,2),next(2,3)},[0.22693135523592908]:{action(bake,1),status(dry),-holds(holding,1),-holds(holding,2),step(2),status(holding),action(pick,2),success,goal,time(1),time(2),time(3),holds(holding,3),step(1),holds(dry,1),holds(dry,3),next(1,2),next(2,3),holds(dry,2)}}